packages/py3_pyparsing.rb

```diff
@@ -3,33 +3,29 @@
 class Py3_pyparsing < Package
   description 'The pyparsing module is an alternative approach to creating and executing simple grammars, vs. the traditional lex/yacc approach, or the use of regular expressions.'
   homepage 'https://github.com/pyparsing/pyparsing/'
-  @_ver = '2.4.7'
-  version "#{@_ver}-1"
+  @_ver = '3.0.9'
+  version @_ver
   license 'MIT'
   compatibility 'all'
   source_url 'https://github.com/pyparsing/pyparsing.git'
   git_hashtag "pyparsing_#{@_ver}"
 
   binary_url({
-    aarch64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/py3_pyparsing/2.4.7-1_armv7l/py3_pyparsing-2.4.7-1-chromeos-armv7l.tpxz',
-     armv7l: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/py3_pyparsing/2.4.7-1_armv7l/py3_pyparsing-2.4.7-1-chromeos-armv7l.tpxz',
-       i686: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/py3_pyparsing/2.4.7-1_i686/py3_pyparsing-2.4.7-1-chromeos-i686.tpxz',
-     x86_64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/py3_pyparsing/2.4.7-1_x86_64/py3_pyparsing-2.4.7-1-chromeos-x86_64.tpxz'
+    i686: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/py3_pyparsing/3.0.9_i686/py3_pyparsing-3.0.9-chromeos-i686.tar.zst',
+  x86_64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/py3_pyparsing/3.0.9_x86_64/py3_pyparsing-3.0.9-chromeos-x86_64.tar.zst'
   })
   binary_sha256({
-    aarch64: '5c84dec013f9409d5ddad79f5c4034c10541926dde48d5848d946284b2aacb12',
-     armv7l: '5c84dec013f9409d5ddad79f5c4034c10541926dde48d5848d946284b2aacb12',
-       i686: 'c084beab43308489cdc1f2814776354e5d2b94c680e7c75ceccf330945747659',
-     x86_64: '10addc1059f3399f556b9ee9b6b2206abf8c0131ec74ec853465032274acbce9'
+    i686: 'c0b44b7b246de089e25f55dbea208910a4489565af2f0d694435fdd49aad647a',
+  x86_64: 'a198eb924014106ccb4577cd3bb4d55be6123c303eb4a445535d2cde488ea694'
   })
 
-  depends_on 'py3_setuptools' => :build
+  depends_on 'py3_pyparsing'
 
   def self.build
-    system "python3 setup.py build #{PY3_SETUP_BUILD_OPTIONS}"
+    system "python3 -m build #{PY3_BUILD_OPTIONS}"
   end
 
   def self.install
-    system "python3 setup.py install #{PY_SETUP_INSTALL_OPTIONS}"
+    system "python3 -m installer #{PY3_INSTALLER_OPTIONS}"
```

Comment:
My attempt to build the arm binary resulted in the following:
```
py3_pyparsing: The pyparsing module is an alternative approach to creating and executing simple grammars, vs. the traditional lex/yacc approach, or the use of regular expressions.
No precompiled binary available for your platform, downloading source...
hint: Using 'master' as the name for the initial branch. This default branch name
hint: is subject to change. To configure the initial branch name to use in all
hint: of your new repositories, which will suppress this warning, call:
hint: 
hint: 	git config --global init.defaultBranch <name>
hint: 
hint: Names commonly chosen instead of 'master' are 'main', 'trunk' and
hint: 'development'. The just-created branch can be renamed via this command:
hint: 
hint: 	git branch -m <name>
Initialized empty Git repository in /usr/local/tmp/crew/py3_pyparsing.20220531051234.dir/.git/
remote: Enumerating objects: 173, done.
remote: Counting objects: 100% (173/173), done.
remote: Compressing objects: 100% (166/166), done.
remote: Total 173 (delta 4), reused 71 (delta 1), pack-reused 0
Receiving objects: 100% (173/173), 970.91 KiB | 6.84 MiB/s, done.
Resolving deltas: 100% (4/4), done.
From https://github.com/pyparsing/pyparsing
 * tag               pyparsing_3.0.9 -> FETCH_HEAD
HEAD is now at 01969ff Back out Lox language parser (from Crafting Interpreters, by Robert Nystrom)
Repository downloaded.
Building from source, this may take a while...
/usr/local/bin/python3: No module named build
/usr/local/lib/crew/lib/package.rb:258:in `system': Command failed with exit 1: bash
py3_pyparsing failed to build: `CFLAGS="-O2 -pipe -flto -ffat-lto-objects -fPIC -fuse-ld=mold " CXXFLAGS="-O2 -pipe -flto -ffat-lto-objects -fPIC -fuse-ld=mold " FCFLAGS="-O2 -pipe -flto -ffat-lto-objects -fPIC -fuse-ld=mold " FFLAGS="-O2 -pipe -flto -ffat-lto-objects -fPIC -fuse-ld=mold " LDFLAGS="-flto " bash -c python3 -m build --wheel --no-isolation` exited with 1
```

Reply:
this build assumes that there is a previous binary version of `py3_build` available on the system. Unforunately this isn't the case for the very first time so you have to use pip, as I said in the initial comment `pip install flit-core build installer`. Unfortunately there isn't another way around this, as it seems like setup.py is being phased out.